lib/linguist/heuristics.rb

```diff
@@ -400,6 +400,14 @@ def call(data)
       end
     end
 
+    disambiguate ".q" do |data|
+      if /[A-Za-z\.][A-Za-z0-9\.]*:{/.match(data) || /(^|\n)\\(cd?|d|l|p|ts?) /.match(data)
+        Language["q"]
+      else
```

Comment:
Couldn't we restrict this to files that contain a line starting with `--` (since you mentioned these are comments in HiveQL)?

Reply:
Also, this is basically SQL. We could probably use a number of heuristic rules specific to SQL (e.g., `SELECT \*` or `ALTER (table|database)`).

Reply:
I took the second approach.